packages/react-components/src/AddressSmall.tsx

```diff
@@ -23,94 +32,174 @@ interface Props {
+function AddressSmall ({ children,
+  className = '',
+  defaultName,
+  isAzeroIdShown = false,
+  isParentAddressShown = false,
+  isRegisterLinkShown = false,
+  onClickName,
+  overrideName,
+  parentAddress,
+  toggle,
+  value,
+  withShortAddress = false,
+  withSidebar = true }: Props): React.ReactElement<Props> {
+  const { queueAction } = useQueue();
+  const { t } = useTranslation();
+
+  const onCopy = useCallback(
+    (address: string) => queueAction({
+      account: address,
+      action: t<string>('clipboard'),
+      message: t<string>('account address copied'),
+      status: 'queued'
+    }),
+    [queueAction, t]
+  );
+
   return (
-    <StyledDiv className={`${className} ui--AddressSmall ${(parentAddress || withShortAddress) ? 'withPadding' : ''}`}>
-      <span className='ui--AddressSmall-icon'>
-        <IdentityIcon value={value as Uint8Array} />
-      </span>
-      <span className='ui--AddressSmall-info'>
-        {parentAddress && (
-          <div className='parentName'>
-            <ParentAccount address={parentAddress} />
-          </div>
-        )}
-        <AccountName
-          className={`accountName ${withSidebar ? 'withSidebar' : ''}`}
-          defaultName={defaultName}
-          onClick={onClickName}
-          override={overrideName}
-          toggle={toggle}
-          value={value}
-          withSidebar={withSidebar}
+    <Container
+      className={className}
+    >
+      {isParentAddressShown && (
+        <div className='parentName'>
+          {parentAddress && (
+            <StyledParentAccount
+              address={parentAddress}
+            />
+          )}
+        </div>
+      )}
+      <StyledIdentityIcon
+        className='identityIcon'
+        value={value as Uint8Array}
+      />
+      <StyleAccountName
+        className={`accountName ${withSidebar ? 'withSidebar' : ''}`}
+        defaultName={defaultName}
+        onClick={onClickName}
+        override={overrideName}
+        toggle={toggle}
+        value={value}
+        withSidebar={withSidebar}
+      >
+        {children}
+      </StyleAccountName>
+      {withShortAddress && (
+        <CopyToClipboard
+          onCopy={onCopy}
+          text={value?.toString() || ''}
         >
-          {children}
-        </AccountName>
-        {withShortAddress && (
-          <div
-            className='shortAddress'
-            data-testid='short-address'
-          >
-            {value}
-          </div>
-        )}
-      </span>
-    </StyledDiv>
+          <AddressContainer className='shortAddress'>
+            <ShortAddress
+              data-testid='short-address'
+            >
+              {value?.toString()}
+            </ShortAddress>
+            <SmallIcon
+              icon='copy'
+            />
+          </AddressContainer>
+        </CopyToClipboard>
+      )}
+      {isAzeroIdShown && (
+        <div className='azeroIdDomain'>
```

Comment:
Not going with SC?

Reply:
I wanted to have all grid related styles in one place for readability, hence this component:
```
const Container = styled.div`
  padding-block: 0.75rem;

  display: grid;
  grid-template-columns: max-content 1fr;
  grid-template-areas:
    "     .        parentName  "
    "identityIcon  accountName "
    "     .       shortAddress "
    "     .       azeroIdDomain";

  align-items: center;
  column-gap: 0.5rem;

  .parentName {
    grid-area: parentName;
    height: 18px;
  }

  .identityIcon {
    grid-area: identityIcon;
  }

  .accountName {
    grid-area: accountName;
  }

  .shortAddress {
    grid-area: shortAddress;
    height: 18px;
  }

  .azeroIdDomain {
    grid-area: azeroIdDomain;
    height: ${AZERO_ID_ROW_HEIGHT};
  }
`;
```

Reply:
Ok makes sense 👍 
I was tackling the same problem the other day and came up with [such "compound" component](https://github.com/Cardinal-Cryptography/common-frontend-research/blob/05193e20d9bd38210d12d57ac218446a0680e698/src/components/PoolDetails/Grid.tsx#L13-L47) (a set of components composing the grid, isolated by a module, so that consumers can't mess it up) - just FYI, not sure if it makes sense in here.